lib/iris/coords.py

```diff
@@ -1478,32 +1619,74 @@ def _repr_other_metadata(self):
             result += ', circular=%r' % self.circular
         return result
 
-    @property
-    def points(self):
-        """The local points values as a read-only NumPy array."""
-        return self._points
-
-    @points.setter
-    def points(self, points):
-        points = np.array(points, ndmin=1)
```

Comment:
Additionally, if the new value doesn't have attributes `ndim` and `dtype` (e.g., if it's a Python list) , `_new_points_requirements` will fail with an `AttributeError`.